addons/mail/static/tests/thread/attachment_list.test.js

```diff
@@ -68,6 +68,49 @@ test("layout with card details and filename and extension", async () => {
     await contains(".o-mail-AttachmentCard small", { text: "txt" });
 });
 
+test("link-type attachment should have open button instead of download button", async () => {
+    const pyEnv = await startServer();
+    const channelId = pyEnv["discuss.channel"].create({
+        channel_type: "channel",
+        name: "channel1",
+    });
+    const attachment_ids = pyEnv["ir.attachment"].create([
+        {
+            name: "url.example",
+            mimetype: "text/plain",
+            type: "url",
+            url: "https://www.odoo.com",
+        },
+        {
+            name: "test.txt",
+            mimetype: "text/plain",
+        },
+    ]);
+    pyEnv["mail.message"].create({
+        attachment_ids,
+        body: "<p>Test</p>",
+        model: "discuss.channel",
+        res_id: channelId,
+        message_type: "comment",
+    });
+    await start();
+    await openDiscuss(channelId);
+    await contains(".o-mail-AttachmentCard", { count: 2 });
+    await contains(":nth-child(1 of .o-mail-AttachmentCard)", { text: "url.example" });
+    await contains(":nth-child(2 of .o-mail-AttachmentCard)", { text: "test.txt" });
+    await contains(
+        ":nth-child(1 of .o-mail-AttachmentCard) .o-mail-AttachmentCard-aside a[title='Open Link']"
+    );
+    await contains(
+        ":nth-child(1 of .o-mail-AttachmentCard) .o-mail-AttachmentCard-aside button[title='Download']",
+        { count: 0 }
+    );
+    await contains(
+        ":nth-child(2 of .o-mail-AttachmentCard) .o-mail-AttachmentCard-aside button[title='Download']"
+    );
+    await contains(`.o-mail-AttachmentCard-aside a[title='Open Link'][target='_blank']`);
```

Comment:
Can replace `:nth-child` to `eq()`, I didn't use in `17` because the `@mail/contains` was not supporting HOOT selector so I had to put native selector.